agent/agent.go

```diff
@@ -26,10 +25,14 @@ type Agent struct {
 }
 
 type Config struct {
-	InstanceName    string
-	SSHAddress      string
-	DiscoveryPeriod time.Duration
-	CollectorConfig *collector.Config
+	InstanceName                string
+	SSHAddress                  string
+	ClusterDiscoveryPeriod      time.Duration
```

Comment:
I'd encapsulate the specific `ClusterDiscoveryPeriod`, `SAPSystemDiscoveryPeriod` and so forth under a `DiscoveryPeriodConfig` or something similar.